File: Interviews/LeetCode/Daily-Problems/2022/October/Day-12/976.py
class Solution:
    def check(self, a, b, c) -> bool:
        if a + b > c and a + c > b and b + c > a:
            return True
        return False
        
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        for i in range(len(nums) - 1, 1, -1):
            if self.check(nums[i], nums[i - 1], nums[i - 2]):
                return nums[i] + nums[i - 1] + nums[i - 2]
        
        return 0

File: Interviews/LeetCode/Daily-Problems/2022/October/Day-12/976.cpp
class Solution {
public:
    bool check(int a, int b, int c) {
        return (a + b > c and a + c > b and b + c > a);
    }
    int largestPerimeter(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        for (int i = nums.size() - 1; i > 1; i--) {
            if (check(nums[i], nums[i - 1], nums[i - 2])) {
                return nums[i] + nums[i - 1] + nums[i - 2];
            }
        }

        return 0;
    }
};